x/ethbridge/keeper/keeper_test.go

```diff
@@ -0,0 +1,255 @@
+		ethereumSender,
+		cosmosReceivers[0],
+		validator1Pow3,
+		amount,
+		claimType,
+	)
+
+	status, err := keeper.ProcessClaim(ctx, ethBridgeClaim)
+
+	require.NoError(t, err)
+	require.Equal(t, status.Text, oracle.PendingStatusText)
+
+	status, err = keeper.ProcessClaim(ctx, ethBridgeClaim)
+	require.Error(t, err)
+	require.True(t, strings.Contains(err.Error(), "already processed message from validator for this id"))
+
+	// other validator execute
+
+	ethBridgeClaim = types.NewEthBridgeClaim(
+		5777,
+		ethBridgeAddress, // bridge registry
+		nonce,
+		symbol,
+		tokenContractAddress, // loopring
+		ethereumSender,       // accounts[0]
+		cosmosReceivers[0],
+		validator2Pow3,
+		amount,
+		claimType,
+	)
+	status, err = keeper.ProcessClaim(ctx, ethBridgeClaim)
+	require.NoError(t, err)
+	require.Equal(t, status.Text, oracle.SuccessStatusText)
+
+}
+func TestProcessSuccessfulClaimLock(t *testing.T) {
+	ctx, keeper, bankKeeper, _, _, _ := CreateTestKeepers(t, 0.7, []int64{3, 3}, "")
+
+	receiverCoins := bankKeeper.GetCoins(ctx, cosmosReceivers[0])
+	require.Equal(t, receiverCoins, sdk.Coins{})
+
+	claimType, err := types.StringToClaimType("lock")
+	require.NoError(t, err)
+	claimContent := types.NewOracleClaimContent(cosmosReceivers[0], amount, symbol, tokenContractAddress, claimType)
+
+	claimBytes, err := json.Marshal(claimContent)
+	require.NoError(t, err)
+	claimString := string(claimBytes)
+	err = keeper.ProcessSuccessfulClaim(ctx, claimString)
+	require.NoError(t, err)
+
+	receiverCoins = bankKeeper.GetCoins(ctx, cosmosReceivers[0])
+
+	require.Equal(t, receiverCoins.String(), "10cstake")
+
+	// duplicate processSuccessClaim
+	err = keeper.ProcessSuccessfulClaim(ctx, claimString)
+	require.NoError(t, err)
+
+	//receiverCoins = bankKeeper.GetCoins(ctx, cosmosReceivers[0])
+	//require.Equal(t, "10cstake", receiverCoins.String())
+	//actually is 20, could this be infinite mint?
+}
+
+func TestProcessSuccessfulClaimBurn(t *testing.T) {
+	ctx, keeper, bankKeeper, _, _, _ := CreateTestKeepers(t, 0.7, []int64{3, 3}, "")
+
+	receiverCoins := bankKeeper.GetCoins(ctx, cosmosReceivers[0])
+	require.Equal(t, receiverCoins, sdk.Coins{})
+
+	claimType, err := types.StringToClaimType("burn")
+	require.NoError(t, err)
+	claimContent := types.NewOracleClaimContent(cosmosReceivers[0], amount, symbol, tokenContractAddress, claimType)
+
+	claimBytes, err := json.Marshal(claimContent)
+	require.NoError(t, err)
+	claimString := string(claimBytes)
+	err = keeper.ProcessSuccessfulClaim(ctx, claimString)
+	require.NoError(t, err)
+
+	receiverCoins = bankKeeper.GetCoins(ctx, cosmosReceivers[0])
+
+	require.Equal(t, receiverCoins.String(), "10stake")
+
+	// duplicate processSuccessClaim
+	err = keeper.ProcessSuccessfulClaim(ctx, claimString)
+	require.NoError(t, err)
+
+	//receiverCoins = bankKeeper.GetCoins(ctx, cosmosReceivers[0])
+	//require.Equal(t, "10stake", receiverCoins.String())
+	//actually is 20, could this be infinite mint?
+}
```

Comment:
Check to see if this behavior is correct or not.  @juniuszhou , could you comment?  If so, remove comment.  If behavior is not correct or represents work in progress, then comment should reflect that.

Reply:
Yes, we can mint the coin lots of time if call the ProcessSuccessfulClaim directly. For external interface, we have the OracleID to prevent duplicated claim. The behavior is correcct.